Hyper-V: Added 2 VHDx Resize tests

santoshx opened the pull request:
VHDX-RESIZE-GROW-FILESYSTEM-512
VHDX-RESIZE-GROW-FILESYSTEM-4096

santoshx commented:
Test Results Summary
ICA test run on 10/24/2018 19:12:15
VHD under test disco.vhdx
ARM Image under test  :  :  : 
Total Executed TestCases        2 (       2 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:27
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. VHDX-RESIZE-GROW-FILESYSTEM-512 : PASS : 14.15
2. VHDX-RESIZE-GROW-FILESYSTEM-4096 : PASS : 13.36

chvalean commented on Testscripts/Linux/STOR_VHDXResize_GrowFSAfterResize.sh:
```diff
@@ -0,0 +1,152 @@
+    # Mount partition
+    if [ ! -e "$mntDir" ]; then
+        mkdir $mntDir
+		check_exit_status "Create mount point" "LogMsg"
```
fix indentation

chvalean commented on Testscripts/Linux/STOR_VHDXResize_GrowFSAfterResize.sh:
```diff
@@ -0,0 +1,152 @@
+    fi
+
+    mount "$deviceName"1 $mntDir
+	check_exit_status "Mount partition" "LogMsg"
```
fix indentation

santoshx replied:
done

chvalean commented on Testscripts/Linux/STOR_VHDXResize_GrowFSAfterResize.sh:
```diff
@@ -0,0 +1,152 @@
+    # Mount partition
+    if [ ! -e "$mntDir" ]; then
+        mkdir $mntDir
+		check_exit_status "Create mount point" "LogMsg"
```
fix indentation

santoshx replied:
done

chvalean commented on Testscripts/Windows/STOR-VHDX-RESIZE-GROWFS.ps1:
```diff
@@ -0,0 +1,204 @@
+
+	# for IDE & offline need to stop VM before resize
+	if ( $controllerType -eq "IDE" -or $testParameters.Offline -eq "True") {
+		LogMsg "Stopping VM for IDE disk or offline is requested"
```
is required*

santoshx replied:
done

chvalean commented on Testscripts/Windows/STOR-VHDX-RESIZE-GROWFS.ps1:
```diff
@@ -0,0 +1,204 @@
+	# for IDE & offline need to stop VM before resize
+	if ( $controllerType -eq "IDE" -or $testParameters.Offline -eq "True") {
+		LogMsg "Stopping VM for IDE disk or offline is requested"
+		Stop-VM -VMName $vmName -ComputerName $hvServer -force
```
please don't force VM shutdown, a graceful shutdown is expected to work

santoshx replied:
done

chvalean commented on Testscripts/Windows/STOR-VHDX-RESIZE-GROWFS.ps1:
```diff
@@ -0,0 +1,204 @@
+	}
+
+	LogMsg "Check if the guest detects the new space"
+	$sd = "sdc"
```
can you move this at the begging of the script/function?
It would be hard to spot it in the code if needs to be changed later on.

santoshx replied:
It's very local to those lines below it. I feel it should be here only.

chvalean commented on XML/TestCases/FunctionalTests.xml:
```diff
@@ -1370,4 +1370,36 @@
+		<testParameters>
+			<param>SCSI=0,1,Dynamic,512,3GB</param>
+			<param>NewSize=4GB</param>
+			<param>fileSystems=(ext4 ext3 xfs)</param>
```
remove ext3

santoshx replied:
done. It reduced test time by couple of minutes.

chvalean commented on XML/TestCases/FunctionalTests.xml:
```diff
@@ -1370,4 +1370,36 @@
+		<testParameters>
+			<param>SCSI=0,1,Dynamic,4096,3GB</param>
+			<param>NewSize=4GB</param>
+			<param>fileSystems=(ext4 ext3 xfs)</param>
```
remove ext3

santoshx replied:
done

chvalean commented:
@santoshx please fix the conflict in xml so I can merge.
But more important the indentation in it for your tests must be fixed - it uses tabs instead of spaces.

santoshx commented:
> @santoshx please fix the conflict in xml so I can merge.
> But more important the indentation in it for your tests must be fixed - it uses tabs instead of spaces.

Fixed conflict and changed tabs to spaces.
